### ISSUE TITLE
Cannot cast java.lang.Boolean to boolean

### ISSUE THREAD

**Issue by tomalbrc** [NerjalNosk/JsonLight#24]
The following code throws a `ClassCastException: Cannot cast java.lang.Boolean to boolean`
```
public class FilamentConfig {
    public boolean emissive_shader = true;
    public boolean commands = true;
    public boolean trimArmorReplaceChainmail = false;
    public boolean preventAdventureModeDecorationInteraction = true;
}

FileParser fileParser = new FileParser(reader); // reader is just an InputStreamReader
var el = fileParser.parse();
FilamentConfig config = JsonMapper.map(el, FilamentConfig.class);
```
at those lines of code in JsonLight (JsonMapper.java):
```
// bool
else if (target == Boolean.class || target == boolean.class) {
    if (!element.isBoolean()) throw new JsonCastingError(element, target);
    return target.cast(element.getAsBoolean());
}
```

**Comment by NerjalNosk** [comment]
Primitive to wrapper class casting does be somewhat janky. I'll look into making a quick 4.1 patch for that and a few other small things I've fixed since then. Thank you for reporting it

**Comment by NerjalNosk** [comment]
Patched with ef29eb0, will push to Sonatype as soon as I figure out the Sonatype new OSSRH system, or retrieve my token, whichever I am lacking. Should be up on Central by the end of the week
